helix-term/src/commands/typed.rs

```diff
@@ -1527,6 +1527,46 @@ fn tree_sitter_scopes(
     Ok(())
 }
 
+fn tree_sitter_highlight_name(
+    cx: &mut compositor::Context,
+    _args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+
+    let (view, doc) = current!(cx.editor);
+    let text = doc.text().slice(..);
+
+    let pos = doc.selection(view.id).primary().cursor(text);
+
+    let mut highlight_name: &str = "";
+    if let Some(highlight) =
+        helix_core::syntax::get_highlight_for_node_at_position(doc.syntax(), text, pos)
+    {
+        let theme = &cx.editor.theme;
+        highlight_name = theme.scope(highlight.0);
+    }
+
+    let content = format!("``json\n{:?}\n````", highlight_name);
```

Comment:
Looks like the bacticks are unbalanced here which is causing the <code>``json</code> in the screenshots. I don't think we need any fancy formatting here though, let's just turn the scope into a String and use that without syntax highlighting:

```rust
let content = cx.editor.theme.scope(highlight.0);
```